### ISSUE TITLE
[PROPOSAL] Return Header Only on Head Request

### ISSUE THREAD

**Issue by hadihammurabi** [ngamux/ngamux#16]
**Is your feature request related to a problem? Please describe.**
Some conditions, we need to get response header only.
It can do by send request using HEAD method.
But, I can't do it using this library.

**Describe the solution you'd like**
Check HEAD request method for all request.
Remove the body and send header only.

References:
- https://reqbin.com/Article/HttpHead

**Comment by born2ngopi** [comment]
it's change the request body with nil ? @hadihammurabi

**Comment by hadihammurabi** [comment]
> it's change the request body with nil ? @hadihammurabi

Yes, it could be like that.

**Comment by born2ngopi** [comment]
hmm i think go already handle this issue, 

check example : https://go.dev/play/p/P8iaWB8T_CR

i send request with HEAD method and body :
``` json
{
   "id":1
}
```

then you can get the output 

``` bash
$ go run main.go
HEAD method
{} body
EOF error
```

**Comment by hadihammurabi** [comment]
> hmm i think go already handle this issue @hadihammurabi ,
> 
> check example : https://go.dev/play/p/P8iaWB8T_CR
> 
> i send request with HEAD method and body :
> 
> ```json
> {
>    "id":1
> }
> ```
> 
> then i got the output
> 
> ```shell
> $ go run main.go
> HEAD method
> {} body
> EOF error
> ```
> 
> ya, if you are not sure you can change value `http.Request.Body` with `http.NoBody`
> 
> > NoBody is an io.ReadCloser with no bytes
> 
> if asign `nil` to `http.Request.Body` this can be panic `nil pointer deference` if you want to decode request body

Thanks for suggestion, then this issue will solve by using http.NoBody 👍

**Comment by born2ngopi** [comment]
I think go already handler on this [line](https://cs.opensource.google/go/go/+/refs/tags/go1.18:src/net/http/transfer.go;l=126-130;bpv=0;bpt=1) or this [line](https://cs.opensource.google/go/go/+/refs/tags/go1.18:src/net/http/transfer.go;l=126-130;bpv=0;bpt=1) , please check function [noResponseBodyExpected](https://cs.opensource.google/go/go/+/refs/tags/go1.18:src/net/http/transfer.go;l=247-249;bpv=0;bpt=1) @hadihammurabi . cmiiw

**Comment by hadihammurabi** [comment]
> I think go already handler on this [line](https://cs.opensource.google/go/go/+/refs/tags/go1.18:src/net/http/transfer.go;l=126-130;bpv=0;bpt=1) or this [line](https://cs.opensource.google/go/go/+/refs/tags/go1.18:src/net/http/transfer.go;l=126-130;bpv=0;bpt=1) , please check function [noResponseBodyExpected](https://cs.opensource.google/go/go/+/refs/tags/go1.18:src/net/http/transfer.go;l=247-249;bpv=0;bpt=1) @hadihammurabi . cmiiw

Yes, but it isn't handled in Ngamux yet.
Now, if you send HEAD request, it will return "method not allowed".

**Comment by hadihammurabi** [comment]
fixed in cba70ed
